### PR TITLE
Running the test suite on TravisCI

### PR DESCRIPTION
**Pull request by zdennis**
# What does this PR do?

This PR gets this project running on TravisCI.

## Notable things

• Removes `only: master` so this should run on all pushed branches
• Rails 5 migrations are versioned and will not run Rails 3 and 4 migrations out of the box. Rather than re-jigger how the migrations are made this relies on schema loading on CI.
• `rake db:environment:set` and `ActiveRecord::Schema#enable_extension` have been shimmed on Rails 3 and 4 so all Rails versions can use the same process

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,25 +2,36 @@ langauge: ruby
...
+  - rvm: 2.3.7
     gemfile: gemfiles/rails_3.gemfile
+    env: TEST_APP_DIR=spec/app_rails3
+  - rvm: 2.3.7
+    gemfile: gemfiles/rails_4.gemfile
+    env: TEST_APP_DIR=spec/app_rails4
+  - rvm: 2.3.7
+    gemfile: gemfiles/rails_5.gemfile
+    env: TEST_APP_DIR=spec/app_rails5
+  - rvm: 2.5.5
```
Comment (a-warner) [inline .travis.yml]: @zdennis should we run rails 3 and 2.5.5?

Comment (a-warner) [reply .travis.yml]: or does that not work unless you're on rails LTS?

File: README.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
+
 # ModernSearchlogic
 
+[![Build Status](https://travis-ci.org/Genius/modern_searchlogic.svg?branch=master)](https://travis-ci.org/Genius/modern_searchlogic)
+
```
Comment (a-warner) [inline README.md]: 🤗

File: spec/shared/db/schema.rb [hunk 3]
```diff
@@ -42,7 +42,7 @@
 
   create_table "votes", id: :serial, force: :cascade do |t|
     t.integer "voteable_id", null: false
-    t.string "voteable_type", null: false
+    t.string "voteable_type", limit: 255, null: false
```
Comment (a-warner) [inline spec/shared/db/schema.rb]: @zdennis is there a migration that needs to get checked in as part of this?

File: .travis.yml (hunk 1 above)
Comment (zdennis) [reply .travis.yml]: It doesn’t work unless on LTS. :(

File: spec/shared/db/schema.rb (hunk 3 above)
Comment (zdennis) [reply spec/shared/db/schema.rb]: There is not. This change is from `rake db:schema:dump` from Rails 5.

I ran `rake db:schema:dump` in Rails 3 and Rails 4 and they both have a lot more changes so I assumed the schema was made from a Rails 5 dump. 

I can remove this change from the schema. I'd be interested to find out if you dump it locally if it adds it back in (is it a postgres version thing affecting schema dumping?)

Comment (a-warner) [reply spec/shared/db/schema.rb]: _yikes_ lol i have no recollection of how/where i generated this schema dump but i'm surprised and concerned that it's different between rails versions

in any case probably not worth checking in these changes as part of this PR?

File: .travis.yml (hunk 1 above)
Comment (a-warner) [reply .travis.yml]: ok cool

File: spec/shared/db/schema.rb (hunk 3 above)
Comment (zdennis) [reply spec/shared/db/schema.rb]: I've rebased out the  `limit: 255` change on the `voteable_type` column.

**Comment by zdennis** [comment]
Thanks for reviewing @a-warner! Once the build goes green I'll merge this into master.

**Pull request merged**